services/frontend-service/src/ui/components/dialog/ConfirmationDialog.tsx

```diff
@@ -74,34 +74,60 @@ export type PlainDialogProps = {
 /**
  * A dialog that just renders its children. Invoker must take care of all buttons.
  */
+
 export const PlainDialog: React.FC<PlainDialogProps> = (props) => {
-    const { onClose, open, children } = props;
-    const classPrefix = props.center ? 'confirmation' : 'plain';
+    const { onClose, open, children, center, disableBackground, classNames } = props;
+    const classPrefix = center ? 'confirmation' : 'plain';
+    const initialRef: HTMLElement | null = null;
+    const rootRef = React.useRef(initialRef);
+
     React.useEffect(() => {
-        window.addEventListener('keyup', (event) => {
+        if (!open) {
+            return () => {};
+        }
+        const winListener = (event: KeyboardEvent): void => {
             if (event.key === 'Escape') {
                 onClose();
             }
-        });
-        document.addEventListener('click', (event) => {
-            if (open) {
-                if (event.target instanceof HTMLElement) {
-                    const isOutside = event.target.className.indexOf(classPrefix + '-dialog-container') >= 0;
-                    if (isOutside) {
-                        onClose();
-                    }
-                }
+        };
+        const docListener = (event: MouseEvent): void => {
```

Comment:
this may be a tricky case, but please try to write a test for the `docListener`.

Reply:
since tests are not part of this story: Please create the story, and mention the story-id in the PR, e.g.
`// Tests will be added in SRX-1234`